src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3975,7 +3975,7 @@ GenTree* Lowering::LowerHWIntrinsicTernaryLogic(GenTreeHWIntrinsic* node)
 
                         if (!op1->IsCnsVec())
                         {
-                            op1->SetUnusedValue();
+                            node->Op(1)->SetUnusedValue();
```

Comment:
I don't think this is quite the right fix.

Rather I think this resolves the issue but misses that `op1` itself is incorrect at this point so the `if (!op1->IsCnsVec())` is also wrong.

We should either be doing `std::swap(op1, op2);` and similar in the places above we're doing `std::swap(node->Op(1), node->Op(2));`

or we should reaquire op1/op2/op3 from the node prior to the second `switch (useFlags)` after we've done any relevant swapping

This line should then stay as is, since its meant to mark op1 as unused if it weren't a constant and was being replaced

Reply:
I will probably just do the "reacquire" fix for now, since it will be pretty surgical.

Reply:
Sounds good, I think that might be the cleanest fix as well; since it helps avoid future refactorings missing doing a swap or anything similar